packages/shared/src/components/auth/AuthOptionsInner.tsx

```diff
@@ -147,11 +152,34 @@ function AuthOptionsInner({
   );
   const [isRegistration, setIsRegistration] = useState(false);
   const windowPopup = useRef<Window>(null);
-  const onLoginCheck = (shouldVerify?: boolean) => {
+
+  const checkForOnboardedUser = async () => {
+    onAuthStateUpdate({ isLoading: true });
+    const isOnboardingPage = router?.pathname?.startsWith('/onboarding');
+
+    if (isOnboardingPage) {
+      const userActions = await getUserActions();
+      const isUserOnboardingComplete = onboardingMandatoryActions.every(
+        (action) =>
+          userActions.some((userAction) => userAction.type === action),
+      );
+
+      if (isUserOnboardingComplete) {
+        await redirectToApp(router);
+        return true;
+      }
+    }
+
+    onAuthStateUpdate({ isLoading: false });
+    return false;
+  };
```

Comment:
This is the check that we are going to apply on login

Reply:
So basically this whole case is for users that get back to funnel but are already onboarded?

Reply:
Yes, if you click on "Continue with Google" on intro screen of the funnel and you are an old user this check will be called. 
Also happens for social/email login from the "non funnel" screens in onboarding  ✔️ 

I've added on auth options so we don't have to duplicate the logic between funnel/onboarding.

Reply:
Either way, when merged test out [critical flows](https://dailydotdev.atlassian.net/wiki/spaces/HAN/pages/458096761/Critical+flows) in regards to login/registration.

Reply:
@capJavert before this fix the user will run into the funnel again and again after every login